core/parsing_handler.py

```diff
@@ -53,28 +51,12 @@ def parse_and_validate(directory):
     return sequencing_run
 
 
-def run_is_new(directory):
+def get_run_status(directory):
```

Comment:
The parsing_handler now can return the full status object, and the status object has `is_X` methods to determine status. This decouples the `progress_handler` from the `progress` module, and simplifies the coupling that exists between the `progress` module and the  `cli_entry` module.